vfs/mem_fs_test.go

```diff
@@ -227,12 +227,17 @@ func TestList(t *testing.T) {
 		}
 	}
 
+	require.NoError(t, fs.Link("/foo/3", "/bar/link-to-3"))
+	require.NoError(t, fs.Link("/foo/3", "/bar/another-link-to-3"))
+
 	{
 		got := fs.String()
 		const want = `          /
        0    a
             bar/
+       0      another-link-to-3
```

Comment:
We could also print some "node ID" so that there is a visual way to see that certain files are actually hard linked to the same node.